src/site/content/en/blog/read-files/index.md

```diff
@@ -0,0 +1,253 @@
+[`<input type="file">`](https://developer.mozilla.org/docs/Web/HTML/Element/input/file#Browser_compatibility),
+[`File`](https://developer.mozilla.org/docs/Web/API/File#Browser_compatibility),
+[`FileList`](https://developer.mozilla.org/docs/Web/API/FileList#Browser_compatibility)
+
+### Select files with a custom drag-and-drop UI (not recommended) {: #drag-and-drop }
+
+{% Aside 'caution' %}
+  [Selecting files with an `<input type="file">` element](#input) is the recommended
+  approach because browser vendors do a lot of work to ensure that their elements
+  are accessible. However, MDN doesn't currently state how many browsers treat their
+  `<input type="file">` elements as built-in drag-and-drop targets.
+{% endAside %}
+
+```html
+<div id="input"></div>
+<script>
+  const input = document.getElementById('input');
+  if (window.FileList && window.File) {
+    input.addEventListener('dragover', event => {
+      event.stopPropagation();
+      event.preventDefault();
+      event.dataTransfer.dropEffect = 'copy';
+    });
+    input.addEventListener('drop', event => {
+      event.stopPropagation();
+      event.preventDefault();
+      const files = event.dataTransfer.files;
+      for (let i = 0; i < files.length; i++) {
+        const file = files[i];
+        const name = file.name ? file.name : 'NOT SUPPORTED';
+        const type = file.type ? file.type : 'NOT SUPPORTED';
+        console.log(`name: ${name}, type: ${type}`);
+      }
+    }); 
+  }
+</script>
+```
+
+This example lets users select multiple files by dragging and dropping them
+over a custom drag-and-drop target on the page and then logs the name and file type
+of each selected file to the Console.
+
+This is mostly the same as the [Select files with an `<input type="file">` element](#input)
+example. The `dragover` event listener may not seem necessary, but without it
+you'll probably find that the drag-and-drop UX doesn't work as intended.
+[`event.dataTransfer.dropEffect = 'copy'`][dropeffect] enables you instruct the browser to
+visually indicate that the drag-and-drop operation is creating a copy of the file,
+rather than moving the original file to a new location for example. When you do a custom drag-and-drop
+UI you access the selected files from `event.dataTransfer.files` in your `drop` event listener,
+which is a `FileList` object.
+
+{# This example doesn't work as an embed. #}
+
+Check out [Custom drag-and-drop](https://custom-drag-and-drop.glitch.me/)
+for a live demonstration.
+
+Browser compatibility data for the APIs used in this section:
+[`<input type="file">`](https://developer.mozilla.org/docs/Web/HTML/Element/input/file#Browser_compatibility),
+[`File`](https://developer.mozilla.org/docs/Web/API/File#Browser_compatibility),
+[`FileList`](https://developer.mozilla.org/docs/Web/API/FileList#Browser_compatibility),
+[`DataTransfer`](https://developer.mozilla.org/docs/Web/API/DataTransfer#Browser_compatibility)
+
+## Read a file's content {: #read }
```

Comment:
Using FileReader is defintiely the most cross-browser compatible way of reading a file's contents, however we should probably also mention the more modern way where you can now call `file.text()`, `file.arrayBuffer()` and `file.stream()` to read files as well? Currently supported in recent versions of chrome and firefox (and trivially polyfillable)

Reply:
Can you give me the polyfill code for each?

Reply:
And these exist on the `File` interface? I'm not seeing any of them on MDN: https://developer.mozilla.org/en-US/docs/Web/API/File

Reply:
`File` inherits from `Blob`, so these are listed on https://developer.mozilla.org/en-US/docs/Web/API/Blob

The easiest way to polyfill these would be using `Response`:
`new Response(file).text()` is equivalent to `file.text()`, `new Response(file).arrayBuffer()` is equivalent to `file.arrayBuffer()` and `new Response(file).body` is equivalent to `file.stream()`. I think at least all modern browsers implement Response, so that should work pretty much everywhere today.

Reply:
Created #2462 for a reminder to do this later.